server-spi/src/main/java/org/keycloak/storage/group/GroupLookupProvider.java

```diff
@@ -64,5 +65,19 @@ default List<GroupModel> searchForGroupByName(RealmModel realm, String search, I
      */
     Stream<GroupModel> searchForGroupByNameStream(RealmModel realm, String search, Integer firstResult, Integer maxResults);
 
+    /**
+     * Returns the group hierarchy filtered by attribute names and attribute values for the given realm.
+     *
+     * For a matching group node the parent group is fetched by id (with all children) and added to the result stream.
+     * This is done until the group node does not have a parent (root group)
+     *
+     * @param realm Realm.
+     * @param attributes Case sensitive attributes.
+     * @param firstResult First result to return. Ignored if negative or {@code null}.
+     * @param maxResults Maximum number of results to return. Ignored if negative or {@code null}.
+     * @return Stream of root groups that have the given string in their name themself or a group in their child-collection has.
+     * The returned hierarchy contains siblings that do not necessarily have a matching name. Never returns {@code null}.
+     */
+    Stream<GroupModel> searchGroupsByAttributes(RealmModel realm, Map<String, String> attributes, Integer firstResult, Integer maxResults);
```

Comment:
Do we need to have this in the `GroupLookupProvider`? What about `GroupProvider`? If I am not mistaken, eventually, we aim to merge all "special" interfaces into one. The reason being with the new store it doesn't make sense to have the methods split anymore.

Reply:
If you need this method here because you have your implementation of Group storage, feel free to leave this here.

Reply:
In the current interface state, this is correct placement.